### PR TITLE
54/unit tests

### PR DESCRIPTION
**Pull request by JacksonSchoe**

### CODE REVIEW

File: SYSC4806_Project.iml [hunk 1]
```diff
@@ -1,2 +1,120 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4" />
\ No newline at end of file
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" version="4">
```
Comment (MichaelDysart) [inline SYSC4806_Project.iml]: Why have so many things been added?

File: pom.xml [hunk 2]
```diff
@@ -20,6 +20,13 @@
     </properties>
 
     <dependencies>
+        <dependency>
```
Comment (MichaelDysart) [inline pom.xml:23]: Did this actually have to be added? Tests were working before.

File: src/test/java/NumberQuestionTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+public class NumberQuestionTest {
+    NumberQuestion nq1;
+    NumberQuestion nq2;
+    NumberQuestion nq3;
+    NumberQuestion nq4;
+
+    @Before
```
Comment (MichaelDysart) [inline src/test/java/NumberQuestionTest.java:17]: FYI, given the style here, I think this should be a BeforeEach. Cause you don't want tests to be affected when you make changes.

**Review by TaylorAbraham — APPROVED** [review]
Looks good, great job adding unit tests all around. There's not a lot to debate in terms of pattern here, approved.

**Review by MichaelDysart — APPROVED** [review]

File: src/test/java/NumberQuestionTest.java (hunk 3 above)
Comment (JacksonSchoe) [reply src/test/java/NumberQuestionTest.java:17]: What do you mean? Does "before" affect tests when you make changes?

Comment (JacksonSchoe) [reply src/test/java/NumberQuestionTest.java:17]: Discussed offline, its good to merge.

**Pull request merged**